prometheus_hardware_exporter/collector.py

```diff
@@ -109,3 +110,397 @@ def fetch(self) -> List[Payload]:
+    def specifications(self) -> List[Specification]:
+        """Define LSI SAS-2 metric specs."""
+        return [
+            Specification(
+                name="lsi_sas_2_controllers",
+                documentation="Number of LSI SAS-2 controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volumes",
+                documentation="Number of IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ready_ir_volumes",
+                documentation="Number of ready IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_unready_ir_volumes",
+                documentation="Number of unready IR volumes",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ir_volume",  # will append "_info" internally
+                documentation="Shows the information about the integrated RAID volume",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_physical_devices",
+                documentation="Number of physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_ready_physical_devices",
+                documentation="Number of ready physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_unready_physical_devices",
+                documentation="Number of unready physical devices",
+                labels=["controller_id"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_physical_device",  # will append "_info" internally
+                documentation="Shows the information about the physical device",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="lsi_sas_2_enclosure",  # will append "_info" internally
+                documentation="Show the information about the enclosure",
+                metric_class=InfoMetricFamily,
+            ),
+            Specification(
+                name="sas2ircu_command_success",
+                documentation="Indicates if the command is successful or not",
+                metric_class=GaugeMetricFamily,
+            ),
+        ]
+
+    def fetch(self) -> List[Payload]:
+        """Load the LSI SAS-2 controllers related information."""
+        adapters = self.sas2ircu.get_adapters()
```

Comment:
this method needs some refactoring. You can extract the comments as functions.

Reply:
Tried refactoring and reduce repeated code as much as possible.